src/main/scala/pureconfig/WordDelimiter.scala

```diff
@@ -0,0 +1,42 @@
+package pureconfig
+
+trait WordDelimiter {
+  def toTokens(s: String): Seq[String]
```

Comment:
I think @ruippeixotog means that you want to base the behaviour of the `ConfigConvert` on the key instead of the type. `ConfigConvert` works for types, not specific keys. @ruippeixotog the canonical way to achieve this is to use a custom new type that wraps another type for those specific keys and then define a `ConfigConvert` for it which differs from the standard one. For instance, let's say that you have a `case class Conf(n: Int, pos: Int)` and you want to have different codecs for `n` and `pos` because `pos` must be positive. The way to do this is to define `case class Pos(int: Int)`, then change `Conf` to `case class Conf(n: Int, pos: Pos)` and then make a codec for `Pos` which checks the value.
This is what other libraries do, for instance `ScalaCheck`.
Now, the question is more if we want to give to the user the ability to define somehow specific codecs for certain keys. I'm against this because I think using a new type is the way to go but I may change idea if there is a compelling use case.